### ISSUE TITLE
constant message order should be enforced

### ISSUE THREAD

**Issue by Trolldemorted** [signal-csharp/Signal-Windows#8]
Currently, the position *might* change after a restart due to two tasks (receiving and sending) racing for the db, while the UI has to display the message immediately.

Having the UI wait for the db insert to solve the dispute is not an option.

Deferring the db access is also not an option, because the signal server wants its messages confirmed asap, and we want to keep the bookkeeping overhead minimal.

tl;dr:
Constraints:
- We want the UI to display an outgoing message immediately. This implies that the UI thread must be the one deciding the order of messages.
- `onMessages` **must not** return before the messages are saved
- insertion of outgoing messages is deferred

Conclusions:
- receive **must** coordinate insertion with UI

Possible solution(s?):
- UI maintains a queue, receive may invoke insertion and block for completion, dedicated task saves to db and unblocks receive
